cartridges/app_custom_homeworks/cartridge/controllers/Newsletter.js

```diff
@@ -0,0 +1,158 @@
+            if (emailExist) {
+                Transaction.wrap(function () {
+                    const customObject = CustomObjectMgr.createCustomObject('NewsletterSubscription', UUID);
+                    customObject.custom.name = name;
+                    customObject.custom.email = email;
+                    customObject.custom.status = 'NEW';
+                });
+                res.json({
+                    success: true,
+                    redirectUrl: URLUtils.url('Newsletter-Success').toString()
+                });
+            } else {
+                res.json({
+                    error: true,
+                    msg: Resource.msg(email + ' already exist', '', null)
+                });
+            }
+
+        } else {
+            res.json({
+                error: true,
+                msg: Resource.msg('Invalid Email', '', null)
+            });
+        }
+    }
+    next();
+});
+
+/**
+ * Newsletter-Success: Rendering success page
+ * @name Newsletter-Success
+ * @function
+ * @memberof Newsletter
+ * @param {httpparameter} - user name
+ */
+server.get('Success', server.middleware.https, function (req, res, next) {
+    const name = req.form.name;
+    const newsletterForm = server.forms.getForm('newsletterSubscription');
+    res.render('newsletterSubscription/newsletterSuccess', {
+        newsletterForm: newsletterForm
+    });
+    next();
+});
+
+/**
+ * Newsletter-Confirmation: Confirming email and changing custom object
+ * @name Newsletter-Confirmation
+ * @function
+ * @memberof Newsletter
+ * @param {httpparameter} - UUID of custom object
+ */
+server.get('Confirmation', server.middleware.https, function (req, res, next) {
+    const CustomObjectMgr = require('dw/object/CustomObjectMgr');
+
+    const uuid = req.querystring.uuid;
+    const customObject = CustomObjectMgr.getCustomObject('NewsletterSubscription', uuid);
+
+    let message = '';
+
+    if (customObject) {
+        if (customObject.custom.status === 'CONFIRMED') {
+            res.render('newsletterSubscription/newsletterConfirmationSame', {
+                name: customObject.custom.name
+            });
+        } else if (customObject.custom.status === 'SENT') {
+            const Transaction = require('dw/system/Transaction');
+
+            Transaction.wrap(function () {
+                customObject.custom.status = 'CONFIRMED';
+            });
+            res.render('newsletterSubscription/newsletterConfirmationSuccess', {
+                name: customObject.custom.name
+            });
+        }
+    } else {
+        const URLUtils = require('dw/web/URLUtils');
+
+        const actionUrl = URLUtils.abs('Newsletter-Start').toString();
+
+        res.render('newsletterSubscription/newsletterConfirmationError', {
+            actionUrl: actionUrl
+        });
+    }
+
+    next();
+});
+
+function validateEmail(email) {
+    const regex = /^[\w.%+-]+@[\w.-]+\.[\w]{2,6}$/;
+    return regex.test(email);
```

Comment:
Please pay attention to https://documentation.b2c.commercecloud.salesforce.com/DOC1/index.jsp?topic=%2Fcom.demandware.dochelp%2Fcontent%2Fb2c_commerce%2Ftopics%2Fsfra%2Fb2c_validate_form_data.html